src/Pim/Component/Catalog/Model/ProductValueCollectionInterface.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Pim\Component\Catalog\Model;
+
+/**
+ * Business collection interface to handle product values.
+ * It does not extends \IteratorAggregate as indexation could be performed internally
+ * for performance reasons.
+ *
+ * Highly inspired by \Doctrine\Common\Collection.
+ *
+ * @author    Julien Janvier <j.janvier@gmail.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface ProductValueCollectionInterface extends \Countable, \IteratorAggregate
+{
+    /**
+     * Adds an value at the end of the collection.
+     *
+     * @param ProductValueInterface $value The value to add.
+     *
+     * @return bool TRUE is the value has been added. FALSE if it was already there.
+     */
+    public function add(ProductValueInterface $value);
+
+    /**
+     * Clears the collection, removing all values.
+     */
+    public function clear();
+
+    /**
+     * Checks whether an value is contained in the collection.
+     * This is an O(n) operation, where n is the size of the collection.
+     *
+     * @param ProductValueInterface $value The value to search for.
+     *
+     * @return bool TRUE if the collection contains the value, FALSE otherwise.
+     */
+    public function contains(ProductValueInterface $value);
+
+    /**
+     * Checks whether the collection is empty (contains no values).
+     *
+     * @return bool TRUE if the collection is empty, FALSE otherwise.
+     */
+    public function isEmpty();
+
+    /**
+     * Removes the value at the specified index from the collection.
+     *
+     * @param string $key The kex/index of the value to remove.
+     *
+     * @return ProductValueInterface The removed value or NULL, if the collection did not contain the value.
+     */
+    public function removeKey($key);
+
+    /**
+     * Removes the specified value from the collection, if it is found.
+     *
+     * @param ProductValueInterface $value The value to remove.
+     *
+     * @return bool TRUE if this collection contained the specified value, FALSE otherwise.
+     */
+    public function remove(ProductValueInterface $value);
+
+    /**
+     * Checks whether the collection contains an value with the specified key/index.
+     *
+     * @param string|integer $key The key/index to check for.
+     *
+     * @return bool TRUE if the collection contains an value with the specified key/index,
+     *                 FALSE otherwise.
+     */
+    public function containsKey($key);
+
+    /**
+     * Gets the value at the specified key/index.
+     *
+     * @param string $key The key/index of the value to retrieve.
```

Comment:
if fixed `containsKey` above
the indexation is made internally and the indexes can only be strings